src/core/dbus-manager.c

```diff
@@ -2198,23 +2210,36 @@ static int method_disable_unit_files_generic(
         if (r < 0)
                 return r;
 
-        r = sd_bus_message_read(message, "b", &runtime);
-        if (r < 0)
-                return r;
+        if (sd_bus_message_is_method_call(message, NULL, "DisableUnitFilesWithFlags")) {
+                r = sd_bus_message_read(message, "t", &flags);
+                if (r < 0)
+                        return r;
+        } else {
+                int runtime;
+
+                r = sd_bus_message_read(message, "b", &runtime);
+                if (r < 0)
+                        return r;
+                flags = unit_file_bools_to_flags(runtime, false);
```

Comment:
doesn't matter, but I think add a comment `/* force = */` before the `false` would be helpful

Reply:
added